### PR TITLE
Revert "Remove unmaintained okd-scos jobs config (#55826)"

### PR DESCRIPTION
**Pull request by sdodson**
This reverts commit fb65ac70578de68dc58b0f0dfa221861d2ec46f9.

It's not clear why these tests are being run all the time and why they're required given at least the ones we've reviewed all look to be `always_run: false` and `optional: true`. So that's worth looking into, but also we expect that these should be fixed via https://github.com/openshift-eng/ocp-build-data/pull/5285

So add them back in hoping to fix things up promptly. We can discuss further on slack if that doesn't come to fruition.

### CODE REVIEW

**Comment by sdodson** [comment]
/pj-rehearse pull-ci-openshift-ironic-image-release-4.17-okd-scos-images
Lets see if https://github.com/openshift-eng/ocp-build-data/pull/5285 fixed things

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16__okd-scos.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-9-golang-1.21-openshift-4.16
+images:
+- build_args:
+  - name: TAGS
+    value: scos
+  dockerfile_path: Dockerfile.ocp
```
Comment (Prashanth684) [inline ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16__okd-scos.yaml:15]:
```suggestion
  dockerfile_path: Dockerfile.okd
```

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.17__okd-scos.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-9-golang-1.22-openshift-4.17
+images:
+- build_args:
+  - name: TAGS
+    value: scos
+  dockerfile_path: Dockerfile.ocp
```
Comment (Prashanth684) [inline ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.17__okd-scos.yaml:15]:
```suggestion
  dockerfile_path: Dockerfile.okd
```

File: ci-operator/config/openshift/ironic-image/openshift-ironic-image-master__okd-scos.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-9-golang-1.22-openshift-4.18
+images:
+- build_args:
+  - name: TAGS
+    value: scos
+  dockerfile_path: Dockerfile.ocp
```
Comment (Prashanth684) [inline ci-operator/config/openshift/ironic-image/openshift-ironic-image-master__okd-scos.yaml:15]:
```suggestion
  dockerfile_path: Dockerfile.scos
```

File: ci-operator/config/openshift/ironic-image/openshift-ironic-image-release-4.16__okd-scos.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-9-golang-1.21-openshift-4.16
+images:
+- build_args:
+  - name: TAGS
+    value: scos
+  dockerfile_path: Dockerfile.ocp
```
Comment (Prashanth684) [inline ci-operator/config/openshift/ironic-image/openshift-ironic-image-release-4.16__okd-scos.yaml:15]:
```suggestion
  dockerfile_path: Dockerfile.scos
```

File: ci-operator/config/openshift/ironic-image/openshift-ironic-image-release-4.17__okd-scos.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-9-golang-1.22-openshift-4.17
+images:
+- build_args:
+  - name: TAGS
+    value: scos
+  dockerfile_path: Dockerfile.ocp
```
Comment (Prashanth684) [inline ci-operator/config/openshift/ironic-image/openshift-ironic-image-release-4.17__okd-scos.yaml:15]:
```suggestion
  dockerfile_path: Dockerfile.scos
```

**Comment by Prashanth684** [comment]
thanks @sdodson . could you also change the dockerfile path in `ci-operator/config/openshift/ironic-image/openshift-ironic-image-master__okd-scos.yaml`  to `Dockerfile.scos` while we're reverting this?

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.17__okd-scos.yaml (hunk 2 above)
Comment (sdodson) [reply ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.17__okd-scos.yaml:15]: okd or scos? why the difference in some of your suggestions?

Comment (Prashanth684) [reply ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.17__okd-scos.yaml]: the [ironic-agent-image](https://github.com/openshift/ironic-agent-image/tree/main) is a different repo from the [ironic-image](https://github.com/openshift/ironic-image)

**Comment by Prashanth684** [comment]
/pj-rehearse pull-ci-openshift-ironic-image-release-4.17-okd-scos-images

**Comment by Prashanth684** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-okd-scos-images

**Comment by Prashanth684** [comment]
the ironic-image job is failing due to: 
```
[2/2] STEP 20/25: COPY ironic-inspector-config/ironic-inspector.conf.j2 /etc/ironic-inspector/
error: build error: building at STEP "COPY ironic-inspector-config/ironic-inspector.conf.j2 /etc/ironic-inspector/": checking on sources under "/tmp/build/inputs": copier: stat: "/ironic-inspector-config/ironic-inspector.conf.j2": no such file or directory
```
it looks like this is because recently, ironic-inspector was [removed](https://github.com/openshift/ironic-image/commit/b38b61f1b47683e6af70325a3414b83e9571f10a#diff-ccb2b3338f8a3382187eabf03c2fe916d8f7be977536228f77eb68d78cc27d79) and `Dockerfile.ocp`  has been fixed to remove the references but not the fcos and scos dockerfiles.

**Comment by elfosardo** [comment]
/pj-rehearse pull-ci-openshift-ironic-image-release-4.17-okd-scos-images pull-ci-openshift-ironic-image-master-okd-scos-images pull-ci-openshift-ironic-agent-image-main-okd-scos-images

**Comment by Prashanth684** [comment]
https://github.com/openshift/ironic-image/pull/567 for 4.17

**Comment by sdodson** [comment]
/pj-rehearse pull-ci-openshift-ironic-image-release-4.17-okd-scos-images pull-ci-openshift-ironic-image-master-okd-scos-images pull-ci-openshift-ironic-agent-image-main-okd-scos-images

**Pull request merged**